docker/Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM node:22
+
+WORKDIR /app
+
+COPY package.json ./
+COPY mesh.config.ts ./
+
+RUN npm i
+
+EXPOSE 4000
+
+ENTRYPOINT ["npx", "mesh-serve"]
```

Comment:
Can we avoid using `npx` here? or make sure it's not going to download anything?

If some dependency is needed, we should use multi-phase Docker steps, where step 1 installs deps and builds everything, and the actual image (step 2) is to run it.

Reply:
I actually think we don't need a package.json at all.
With this PR, a mesh.config.ts file won't be needed too.
`npm i graphql @graphql-mesh/serve-cli` for installation then `npx mesh-serve` would be enough. (npx is still needed though to point to `node_modules` under the directory.
Then users can put their own if additional configuration is needed.